File: основы17.09-1/main.py
import math

# 1
"""
x0 = -1.5
xk = 3.5
dx = 0.5
a = -1.25
b = -1.5
c = 0.75

x = x0
while x <= xk + dx / 2:
    if x <= 0:
        print(f"x = {x}, y = ?")
    else:
        y = ((10 ** -2) * b * c) / x + (math.cos((a ** 3) * x))
        print(f"x = {x}, y = {y}")
    x += dx
"""

# 2
"""
x0 = 5.3
xk = 10.3
dx = 0.25
a = 1.35
b = -6.25

x = x0
while x <= xk + dx / 2:
    if x <= 0:
        print(f"x = {x} y = ?")
    else:
        y = (a * x**3) + (math.cos(x**3 - b)**2)
        print(f"x = {x}, y = {y}")
    x += dx
"""

# 3
"""
x0 = -0.75
xk = -2.05
dx = -0.2

x = x0
while x >= xk + dx / 2:
    if x <= 0:
        print(f"x = {x} y = ?")
    else:
        y = ((9 * x) ** 4) + (math.sin(57.2 + x))
        print(f"x = {x}, y = {y}")
    x += dx
"""

# 4
"""
x0 = -0.73
xk = -1.73
dx = -0.1
b = -2

x = x0
while x >= xk + dx / 2:
    if x <= 0:
        print(f"x = {x} y = ?")
    else:
        y = (math.log(x - b) ** (1/2)) / (((b ** 3) - (x ** 3)) ** 3/2) + (math.log(x - b))
        print(f"x = {x}, y = {y}")
    x += dx
"""

# 5
"""
x0 = 1.23
xk = -2.4
dx = -0.3
b = 12.6

x = x0
while x >= xk + dx / 2:
    if x <= 0:
        print(f"x = {x} y = ?")
    else:
        y = (15.28 * math.log(x ** (-3 / 2))) + math.cos((math.log(x) + b))
        print(f"x = {x}, y = {y}")
    x += dx
"""

File: основы17.09-1/main.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    // 1 
    /*
    double x0 = -1.5, xk = 3.5, dx = 0.5;
    double a = -1.25, b = -1.5, c = 0.75;

    for (double x = x0; x <= xk + dx / 2; x += dx) {
        if (x <= 0)
            cout << "x = " << x << ", y = ?" << endl;
        else {
            double y = (0.01 * b * c) / x + cos(pow(a, 3) * x);
            cout << "x = " << x << ", y = " << y << endl;
        }
    }
    return 0;
    */

    // 2
    /*
    double x0 = 5.3, xk = 10.3, dx = 0.25;
    double a = 1.35, b = -6.25;

    for (double x = x0; x <= xk + dx / 2; x += dx) {
        if (x <= 0)
            cout << "x = " << x << " y = ?" << endl;
        else {
            double y = a * pow(x, 3) + pow(cos(pow(x, 3) - b), 2);
            cout << "x = " << x << ", y = " << y << endl;
        }
    }
    return 0;
    */

    // 3
    /*
    double x0 = -0.75, xk = -2.05, dx = -0.2;

    for (double x = x0; x >= xk + dx / 2; x += dx) {
        if (x <= 0)
            cout << "x = " << x << " y = ?" << endl;
        else {
            double y = pow(9 * x, 4) + sin(57.2 + x);
            cout << "x = " << x << ", y = " << y << endl;
        }
    }
    return 0;
    */

    // 4
    /*
    double x0 = -0.73, xk = -1.73, dx = -0.1;
    double b = -2;

    for (double x = x0; x >= xk + dx / 2; x += dx) {
        if (x <= 0)
            cout << "x = " << x << " y = ?" << endl;
        else {
            double y = sqrt(log(x - b)) / pow(pow(b, 3) - pow(x, 3), 1.5) + log(x - b);
            cout << "x = " << x << ", y = " << y << endl;
        }
    }
    return 0;
    */

    // 5
    /*
    double x0 = 1.23, xk = -2.4, dx = -0.3;
    double b = 12.6;

    for (double x = x0; x >= xk + dx / 2; x += dx) {
        if (x <= 0)
            cout << "x = " << x << " y = ?" << endl;
        else {
            double y = 15.28 * log(pow(x, -1.5)) + cos(log(x) + b);
            cout << "x = " << x << ", y = " << y << endl;
        }
    }
    return 0;
    */

    
}